### DISCUSSION TITLE
keri-dev 06/08/23

### DISCUSSION THREAD

**Discussion by m00sey** [WebOfTrust/keri#31] [Meetings]
[https://us06web.zoom.us/rec/share/OA8aA5JJQSLYo4Bc7fTbSqD66pQaPj7SLN4BOVjhykhQ03fKPglqRJxuRnJJidbL.O3l6L17Q5Q14Ort_](https://atpscan.global.hornetsecurity.com/index.php?atp_str=m2e0rlJ_mk58bG8Pe_6zkcKLL9vVcUmpE0Mu6EFQFzc1P4WAeIlTpa5-tH3OMLYiCLfFjKWyaiA3G-7H0fm2I8WOX0WHRb5FwSEKXLmS-CzQf8MF4YYl6_OmduWlMeLtnNioaR8R-4bS6cqd3StzsOq9VrPBRml2T8xeNq81X2SKwymSFVi32hEYNq8c2hrn2GZC1O4LWxwzqqYGn8J0OL93-l8PGaYirvTS6WMIEeqIPLTcK3VXgQskB3cQc3TbH1822PgnREHWZUMDz6bDy6nfGBq_AEsm2Hv9J9WdLSLFYuOYDM-AWEmXzxKu-tHh0M4nj4NvO4TLAShHvxbnN4J8txiyCbacOuoZ_fOG-iGCzS3IQRRQECcuIJiMRbCWcKelkwLxCVdCbJsLXvy6S7fYbATLG81EQ-cITQhd_xHNBophjoC6MHkO1uWfNG7FvS2kIzo6I6-Qp1vQvC2xOKOEeyM6OiO-g5a1LldlkXvnbyMegAA-)
Passcode: t0JkM^S&

Signify-ts - oobi implementation, working on test coverage and UI.
KERIA - Agent, work on credentialing APIs supporting signing at the edge in keripy
keride - Encrypt/decrypt implementation
keripy - updates to protocols for versioning, updates for revocation

did:keri/did:web

did:web:verifiablelei.org:gleif:internal:EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL
`web` can be replaced by keri and a resolver to create a valid did doc from a did:web like did

Where are definitions that map parts of keri into parts of the did doc
https://github.com/WebOfTrust/ietf-did-keri/issues/8
https://github.com/WebOfTrust/ietf-did-keri/issues/2

**Comment by SmithSamuelM** [comment]
did:web:verifiablelei.org:gleif:internal:EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL
did:web:verifiablelei.org:gleif:external:EINmHd5g7iV-UldkkkKyBIH052bIyxZNBn9pq-zNrYoS
did:web:verifiablelei.org:gleif:root:EDP1vHcw_wc4M__Fj53-cJaBnZZASd-aMTaSyWEQ-PC2

**Reply by ghost** [reply]
https://verifiablelei.org/gleif/internal/EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL/did.json
https://verifiablelei.org/gleif/external/EINmHd5g7iV-UldkkkKyBIH052bIyxZNBn9pq-zNrYoS/did.json
https://verifiablelei.org/gleif/root/EDP1vHcw_wc4M__Fj53-cJaBnZZASd-aMTaSyWEQ-PC2/did.json

**Reply by SmithSamuelM** [reply]
@swcurran
Like this because it has the advantages of simplicity of did:web but fixes some of its problems.
did;web problem is not self certifying but keri did:keri is self certifying.  
You can then use the did:web as a discovery vector to get the KEL to do the verifiability that did:web is missing.
Stable resolvable DIDs.
"We don't need no stinking ledgers"
Wayback machine KEL history
Questions. How events affect the did doc.
- Extensions how does a KEL event that affects a did doc
- Publish resources on web server that are tied to did:keri to get document plus signatures
- want to support anoncreds and verifiable credentials post verifiable files

@markussabadello

Anil John interested in something like did:web but verifiable in some way
Others interested in improving did:web with security features of did:keri

What if lose the domain name so doesn't weaken the KERI approach.

**Reply by SmithSamuelM** [reply]
Define the transformation as `did:keri:aid:domain...` => `did:web:domain...:aid`

Example:
did:keri:EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL:verifiablelei.org:gleif:internal

Transform to:
did:web:verifiablelei.org:gleif:internal:EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL

Resolves to:
https://verifiablelei.org/gleif/internal/EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL/did.json

**Reply by stephancurran** [reply]
@SmithSamuelM, I'm afraid you've tagged the wrong Curran.

**Reply by 2byrds** [reply]
@SmithSamuelM I think you meant @swcurran

**Comment by m00sey** [comment]
From @qui-pdm :

> Could a domain-free form such as `did:keri:EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL` still be supported? So for `did:keri:id:...`, either `id` is a CESR code (without dots) or it is a dotted domain?

> Or let the did:web lookup be a hint and reverse it: `did:keri:{SAID}:domain.tld:…`

> Example:
> did:keri:EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL:verifiablelei.org:gleif:internal
>… resolves to:
>https://verifiablelei.org/gleif/internal/EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL/did.cesr

> define the transformation as `did:keri:aid:domain...` => `did:web:domain...:aid`

> Example:
> did:keri:EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL:verifiablelei.org:gleif:internal
> Transform to:
> did:web:verifiablelei.org:gleif:internal:EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL
> Resolves to:
> https://verifiablelei.org/gleif/internal/EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL/did.json

**Comment by m00sey** [comment]
From @jasoncolburne Can we anchor the domain in an anchor/TEL to track control

**Reply by SmithSamuelM** [reply]
Yes

**Comment by SmithSamuelM** [comment]
@markusSabadello We can use metadata such as equivalentId to express that two DIDs identify the same subject.

**Comment by m00sey** [comment]
10:14:27 From Samuel Smith To Everyone:
	https://github.com/WebOfTrust/keria
10:19:38 From Henk van Cann To Everyone:
	https://weboftrust.github.io/WOT-terms/ KERI Suite Search Engine (KERISSE)
10:20:02 From Stephen Curran To Everyone:
	Peter — what’s on the cup you are drinking from. Is that a Tims cup?
10:21:01 From Philip Feairheller To Everyone:
	did:web:verifiablelei.org:gleif:internal:EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL
	did:web:verifiablelei.org:gleif:external:EINmHd5g7iV-UldkkkKyBIH052bIyxZNBn9pq-zNrYoS
	did:web:verifiablelei.org:gleif:root:EDP1vHcw_wc4M__Fj53-cJaBnZZASd-aMTaSyWEQ-PC2
10:21:38 From Peter David McCormick To Everyone:
	@Stephen Yes! 😂
10:25:00 From Rodolfo Miranda To Everyone:
	https://github.com/WebOfTrust/ietf-did-keri/issues
10:36:31 From Peter David McCormick To Everyone:
	Could a domain-free form such as `did:keri:EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL` still be supported? So for `did:keri:id:...`, either `id` is a CESR code (without dots) or it is a dotted domain?
10:37:34 From Peter David McCormick To Everyone:
	Or let the did:web lookup be a hint and reverse it: `did:keri:{SAID}:domain.tld:…`
10:38:18 From Philip Feairheller To Everyone:
	@Peter, that is an good suggestion
10:38:44 From Philip Feairheller To Everyone:
	But to be compatible with did:web, the domain name comes first, followed by the path
10:38:59 From Peter David McCormick To Everyone:
	Example:
	did:keri:EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL:verifiablelei.org:gleif:internal
	
	… resolves to:
	
	https://verifiablelei.org/gleif/internal/EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL/did.cesr
10:39:26 From Stephen Curran To Everyone:
	Or build the super watcher.
10:39:26 From Trent Larson To Everyone:
	For losing control, I’ve liked an approach I’ve seen where the domain is followed by a timestamp, eg. “verifiablelei.org,2023-06-08”. I don’t see that in did:web unfortunately.
10:39:44 From Stephen Curran To Everyone:
	We just don’t have to wait until the super watcher is available.
10:40:43 From Stephen Curran To Everyone:
	@trent — that’s why to declare did:keri to be compatible with did:web — did:keri can support the extra pieces.
10:42:19 From Peter David McCormick To Everyone:
	Define the transformation as `did:keri:said:domain...` => `did:web:domain...:said`
10:42:32 From Alex To Everyone:
	sounds like did keri is a superset of did web
10:43:39 From Rodolfo Miranda To Everyone:
	You can use query parameters in the DID URL
10:44:22 From Stephen Curran To Everyone:
	Query parameters for the core identifier is not helpful.  It doesn’t help with a stable, resolvable identifier.
10:44:50 From Trent Larson To Everyone:
	Reacted to "sounds like did keri..." with 👍
10:45:00 From Stephen Curran To Everyone:
	Reacted to "sounds like did keri..." with 👍
10:46:04 From Peter David McCormick To Everyone:
	Corrected: define the transformation as `did:keri:aid:domain...` => `did:web:domain...:aid`
10:46:13 From Markus Sabadello To Everyone:
	We can use metadata such as equivalentId to express that two DIDs identify the same subject.
10:46:43 From Peter David McCormick To Everyone:
	Example:
	did:keri:EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL:verifiablelei.org:gleif:internal
	
	Transform to:
	did:web:verifiablelei.org:gleif:internal:EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL
	
	Resolves to:
	https://verifiablelei.org/gleif/internal/EFcrtYzHx11TElxDmEDx355zm7nJhbmdcIluw7UMbUIL/did.json
10:53:20 From Jason Colburne To Everyone:
	Have to drop, great and productive meeting everyone
10:58:24 From Markus Sabadello To Everyone:
	Noooo I want there to be many many DID methods ;)
